CMakeLists.txt

```diff
@@ -210,34 +163,24 @@ if(WW3)
     set(WW3_COMP "orion")
   endif()
   message("Build WW3:")
-  message("   run: ${CMAKE_BUILD_TOOL} WW3_PARCOMPN=4 WW3_COMP=${WW3_COMP} ww3_nems")
+  message("   run: ${CMAKE_BUILD_TOOL} WW3_PARCOMPN=4 WW3_COMP=${WW3_COMP} ww3_nemslibonly")
   message("   in:  ${PROJECT_SOURCE_DIR}/WW3/model/esmf")
   message("")
 
   add_custom_target(ww3_nems
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/WW3/model/esmf
-    COMMAND ${CMAKE_BUILD_TOOL} WW3_PARCOMPN=4 WW3_COMP=${WW3_COMP} ww3_nems > make.log 2>&1
+    COMMAND ${CMAKE_BUILD_TOOL} WW3_PARCOMPN=4 WW3_COMP=${WW3_COMP} ww3_nemslibonly > ${PROJECT_BINARY_DIR}/ww3_make.log 2>&1
   )
+
+    set(WW3_LIBS ${PROJECT_SOURCE_DIR}/WW3/model/obj/libww3_multi_esmf.a)
 endif()
 
 ###############################################################################
-### NEMS
+### NEMS (ufs-weather-model-lib)
 ###############################################################################
-configure_file(${PROJECT_SOURCE_DIR}/NEMS/src/ESMFVersionDefine_ESMF_NUOPC.h
-               ${PROJECT_BINARY_DIR}/NEMS/src/ESMFVersionDefine.h
-               COPYONLY)
-# until we remove "../ESMFVersionDefine.h" from ENS_Cpl/ENS_CplComp_ESMFMod_STUB.F90
-configure_file(${PROJECT_SOURCE_DIR}/NEMS/src/ESMFVersionDefine_ESMF_NUOPC.h
-               ${PROJECT_BINARY_DIR}/NEMS/ESMFVersionDefine.h
-               COPYONLY)
-configure_file(${PROJECT_SOURCE_DIR}/NEMS/src/ESMFConvenienceMacros.h
-               ${PROJECT_BINARY_DIR}/NEMS/src/ESMFConvenienceMacros.h
-               COPYONLY)
-
-add_executable(
-    NEMS.exe
+add_library(
```

Comment:
Can this be made into `libnems` consisting solely of the NEMS source code?

`ufs-weather-model-lib` is a collection of `libnems, fv3atm, ww3_nems`
It will simplify the logic under hear as everything is `target_link_libraries` or `target_compile_definitions` to `ufs-weather-model-lib`.

Reply:
Are you suggesting to rename  `ufs-weather-model-lib` to `nems`? I can do that. But I'm not sure I know how to make `ufs-weather-model-lib` a collection of `nems`, `fv3atm`, and optionally `ww3_nems`.

Reply:
yes.
you can create ufs-weather-model-lib from nems, fv3atm etc. same way you created fv3atm from fv3cpl, gfsphys, etc.

Reply:
But `fv3atm`, in addition to depending on `fv3cpl`, `gfsphysics` also contains code (`atmos_model.F90`, `fv3_cap.F90`, ...).  How do I create library without code and then add other libraries as dependencies. I guess I do not understand what you mean by 'collection'? I do not know how to "_create ufs-weather-model-lib from nems, fv3atm_". I can only add `nems` and `fv3atm` as dependencies to `ufs-weather-model-lib`.